### PR TITLE
Fix reading assignment links

### PR DESCRIPTION
**Pull request by Dantemss**
The fragments were being cached before the content was modified.

### CODE REVIEW

File: app/subsystems/content/models/book.rb [hunk 1]
```diff
@@ -50,6 +50,8 @@ def set_ecosystem_title
   end
 
   def reference_view_url
+    raise('Unpersisted Ecosystem') if ecosystem.id.nil?
+
```
Comment (nathanstitt) [inline app/subsystems/content/models/book.rb]: I think you can also do `ecosystem.persisted?` https://api.rubyonrails.org/classes/ActiveRecord/Persistence.html#method-i-persisted-3F

File: app/subsystems/content/models/page.rb [hunk 2]
```diff
@@ -131,6 +145,8 @@ def context_for_feature_ids(feature_ids)
   end
 
   def reference_view_url(book = chapter.book)
+    raise('Unpersisted Page') if id.nil?
+
```
Comment (nathanstitt) [inline app/subsystems/content/models/page.rb]: same, could use `persisted?`

File: app/subsystems/content/models/book.rb (hunk 1 above)
Comment (Dantemss) [reply app/subsystems/content/models/book.rb]: Yeah same deal but I thought using the id made sense since it's the id we care about

Comment (nathanstitt) [reply app/subsystems/content/models/book.rb]: yes, only diff would be if the page was deleted, in which case it'd have an id, but `persisted?` would be false.  I don't think we delete pages tho

**Pull request merged**